lib/evm/git.rb

```diff
@@ -8,24 +8,36 @@ def exist?
       File.exist?(@path)
     end
 
-    def clone(url, branch = nil)
+    def clone(url, branch = nil, commit = nil)
       args = 'clone', url, @path
       args << "--branch=#{branch}" if branch
+     if commit == nil
+        args << '--depth=1'
+      end
       git(*args)
     end
 
-    def pull
+    def pull(commit = nil)
       Dir.chdir(@path) do
-        git 'pull'
+        args = 'pull'
+        args = 'pull', '--depth=2000000' if commit
```

Comment:
This is pretty hacky. Is it really the only way to unshallow the repository? I have a feeling `--unshallow --tags` works, but haven't tested. The change is not fully described in the commit message, so hard to tell.